Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConcepTypeDeleteController.java

```diff
@@ -61,13 +60,10 @@ public String prepareDeleteType(@PathVariable("typeid") String typeid, ModelMap
         model.addAttribute("supertype", type.getSupertypeId());
         // condition to check enable whether to delete the concepttype
         boolean enableDelete = true;
-        List<ConceptList> conceptLists = conceptListManager.getAllConceptLists();
-        for (ConceptList conceptList : conceptLists) {
-            List<ConceptEntry> conceptEntries = conceptManager.getConceptListEntries(conceptList.getConceptListName());
-            for (ConceptEntry conceptEntry : conceptEntries) {
-                if ((conceptEntry.getTypeId() != null) && (conceptEntry.getTypeId()).equals(typeid)) {
-                    enableDelete = false;
-                }
+        List<ConceptEntry> conceptEntries = conceptManager.getConceptEntryByTypeId(typeid);
+        for (ConceptEntry conceptEntry : conceptEntries) {
+            if ((conceptEntry.getTypeId() != null) && (conceptEntry.getTypeId()).equals(typeid)) {
```

Comment:
since you're requesting concepts that have the type=typeid, isn't this always true?